module/project/templates/helper.js

```diff
@@ -957,74 +958,190 @@ module.exports = class ProjectTemplatesHelper {
+                        throw {
+                            message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                            status : HTTP_STATUS_CODE['bad_request'].status
+                        }
+                    }
                     
-                    templateData[0].projectId = "";
+                    let solutiondata=solutionDocument.data;
+                    let templateId=solutiondata.map(function(id){
+                        return id.projectTemplateId
+                    });
+                    templateId=templateId.toString();
+                    if(!templateId){
+                        // throw {
+                        //     message : CONSTANTS.apiResponses.MISSING_PROJECT_TEMPLATE_ID,
+                        //     status : HTTP_STATUS_CODE['bad_request'].status,
+                        //     result:"checking"
+                        // }
+                        return resolve({
+                            success : false,
+                            data : solutiondata,
+                            message : CONSTANTS.apiResponses.MISSING_PROJECT_TEMPLATE_ID
+                        });   
+                    }
+                    //for templateId: is available
+                    let findQuery = {};
+
+                    let validateTemplateId = UTILS.isValidMongoId(templateId);
+                    if( validateTemplateId ) {
+                        findQuery["_id"] = templateId;
+                    }else {
+                        findQuery["externalId"] = templateId;
+                    }
+                    let templateData = await projectTemplateQueries.templateDocument(findQuery,"all",
+                    [
+                        "ratings",
+                        "noOfRatings",
+                        "averageRating",
+                        "parentTemplateId",
+                        "userId",
+                        "createdBy",
+                        "updatedBy",
+                        "createdAt",
+                        "updatedAt",
+                        "__v"
+                    ]);
+    
+                    if ( !templateData.length > 0 ) {
+                        throw {
+                            status : HTTP_STATUS_CODE.bad_request.status,
+                            message : CONSTANTS.apiResponses.PROJECT_TEMPLATE_NOT_FOUND
+                        }
+                        
+                    }
+    
+                    if (templateData[0].tasks && templateData[0].tasks.length > 0) {
+                        templateData[0].tasks = 
+                        await this.tasksAndSubTasks(templateData[0]._id);
+                    }
+    
+                    let result = await _templateInformation(templateData[0])
+                    if( !result.success ) {
+                        return resolve(result);
+                    }
+    
+                    if( !templateData[0].isReusable ) {
```

Comment:
@VISHNUDAS-tunerlabs this block is not required for link. because userId is empty